dev/AppLifecycle/FileActivatedEventArgs.h

```diff
@@ -47,6 +47,24 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
 
         static winrt::Windows::Foundation::IInspectable Deserialize(winrt::Windows::Foundation::Uri const& uri)
         {
+            // Standard serialize has a query of the form: "?ContractId=Windows.File&Verb=<verb>&File=<file_path>"
+            // If the query has the exact same format, then we can deserialize it with simple string operations,
+            // as QueryParsed() can cause issues when file path contains unicode characters or specical characters like &.
+            std::wstring serializedQueryHeader = L"?" + std::wstring(c_contractIdKeyName) + L"=" + std::wstring(c_fileContractId) + L"&Verb=";
+            int headerLen = (int)wcslen(serializedQueryHeader.c_str());
+            if (CompareStringOrdinal(uri.Query().c_str(), headerLen, serializedQueryHeader.c_str(), -1, TRUE) == CSTR_EQUAL)
+            {
+                int queryLen = (int)wcslen(uri.Query().c_str());
+                const std::wstring query = uri.Query().c_str();
+                int verbEnd = query.find(L"&", headerLen);
+                auto verb = winrt::to_hstring(query.substr(headerLen, verbEnd - headerLen).c_str());
+
+                int filePos = query.find(L"&File=");
```

Comment:
need error checking for [std::string::npos](https://en.cppreference.com/w/cpp/string/basic_string/npos)